crates/hir-ty/src/utils.rs

```diff
@@ -264,12 +279,11 @@ impl Generics {
 
     fn find_param(&self, param: TypeOrConstParamId) -> Option<(usize, &TypeOrConstParamData)> {
         if param.parent == self.def {
-            let (idx, (_local_id, data)) = self
-                .params
-                .iter()
-                .enumerate()
-                .find(|(_, (idx, _))| *idx == param.local_id)
-                .unwrap();
+            let (idx, (_local_id, data)) =
+                match self.params.iter().enumerate().find(|(_, (idx, _))| *idx == param.local_id) {
+                    None if self.is_gat => return None,
+                    val => val.unwrap(),
```

Comment:
I don't know why we have `find_param` return an option if we never actually return `None`. Let's just change it to return `None` if the param isn't found in all cases, and get rid of `is_gat`?